python/lib/sift_py/upload/csv.py

```diff
@@ -0,0 +1,149 @@
+import json
+import mimetypes
+import re
+from pathlib import Path
+from typing import Optional, Tuple, Union
+from urllib.parse import urljoin, urlparse
+
+import requests
+from sift_py.rest import SiftRestConfig
+from sift_py.upload.config import CsvConfig
+from sift_py.upload.status import DataImportStatus
+
+
+class CsvUploadService:
+    UPLOAD_PATH = "/api/v1/data-imports:upload"
+    URL_PATH = "/api/v1/data-imports:url"
+
+    _rest_conf: SiftRestConfig
+    _upload_uri: str
+    _url_uri: str
+    _apikey: str
+
+    def __init__(self, rest_conf: SiftRestConfig):
+        self._rest_conf = rest_conf
+        self._apikey = rest_conf["apikey"]
+        base_uri = self.__class__._compute_uri(rest_conf)
+        self._upload_uri = urljoin(base_uri, self.UPLOAD_PATH)
+        self._url_uri = urljoin(base_uri, self.URL_PATH)
+
+    def upload(
+        self,
+        path: Union[str, Path],
+        csv_config: CsvConfig,
+    ) -> DataImportStatus:
+        posix_path = Path(path) if isinstance(path, str) else path
+
+        if not posix_path.is_file():
+            raise Exception(f"Provided path, '{path}', does not point to a regular file.")
+
+        _, mimetype, content_encoding = self.__class__._mime_and_content_type_from_path(posix_path)
```

Comment:
Just a thought -- since we're getting the MIME, maybe it'd be nice to raise a warning if we detect a type that wouldn't play well with everything downstream of here? e.g. a pdf instead of a csv or something. Just a nit/thought though